src/hammer-vlsi/synthesis/yosys/__init__.py

```diff
@@ -68,88 +119,256 @@ def tool_config_prefix(self) -> str:
+        clock_port = self.get_clock_ports()[0]
+        self.clock_port_name = clock_port.name
+        time_unit = "ps" # yosys requires time units in ps
+        self.clock_period = int(clock_port.period.value_in_units(time_unit))
+        self.clock_uncertainty = int(clock_port.period.value_in_units(time_unit))
+        self.clock_transition = 0.15 # SYNTH_CLOCK_TRANSITION
+
+        self.synth_cap_load = 33.5 # SYNTH_CAP_LOAD
+        self.max_fanout = 5 # default SYNTH_MAX_FANOUT = 5
+
+        # Yosys commands only take a single lib file
+        #   so use typical corner liberty file
+        corners = self.get_mmmc_corners()  # type: List[MMMCCorner]
+        corner_tt = next((corner for corner in corners if corner.type == MMMCCornerType.Extra), None)
+        self.liberty_file = self.get_timing_libs(corner_tt)
+        
+        self.append("yosys -import")
+
+        # replaces undef (x) constants with defined (0/1) constants.
+        # self.append("setundef -zero")
+
+        # We are switching working directories and Yosys still needs to find paths.
+        abspath_input_files = list(map(lambda name: os.path.join(os.getcwd(), name), self.input_files))  # type: List[str]
+        # If we are in hierarchical, we need to remove hierarchical sub-modules/sub-blocks.
+        if self.hierarchical_mode.is_nonleaf_hierarchical():
+            abspath_input_files = list(map(self.remove_hierarchical_submodules_from_file, abspath_input_files))
+        # Add any verilog_synth wrappers (which are needed in some technologies e.g. for SRAMs) which need to be
+        # synthesized.
+        abspath_input_files += self.technology.read_libs([
+            hammer_tech.filters.verilog_synth_filter
+        ], hammer_tech.HammerTechnologyUtils.to_plain_item)
+        
+        for verilog_file in abspath_input_files:
+            self.append(f"read_verilog -sv {verilog_file}")
+
+        liberty_files = self.technology.read_libs([hammer_tech.filters.timing_lib_with_ecsm_filter], hammer_tech.HammerTechnologyUtils.to_plain_item)
+        for lib_file in liberty_files:
+            self.append(f"read_liberty -lib {lib_file}")
+
         return True
 
     def syn_generic(self) -> bool:
-        # TODO: currently using OpenROAD's default synthesis script
+        self.block_append(f"""
+        # TODO: verify this command, it was in yosys manual but not in OpenLANE script
+        yosys proc
+        hierarchy -check -top {self.top_module}
+
+        # old: synth -top {self.top_module}
+        # but techmap file location that yosys is checking for is incorrect
+        # TODO: figure out why this is
+
+        # begin, coarse labels: automatic
+        synth -top {self.top_module} -run :fine
+        # fine label: run manually 
+        opt -fast -full
+        memory_map
+        opt -full
+        # TODO: figure out why techmap path assumed correct for conda but not server install 
+        techmap -map "/usr/local/share/yosys/techmap.v"
```

Comment:
What is this file?

Reply:
It defines internal mapping of logic cells (see [here](https://github.com/YosysHQ/yosys/blob/master/techlibs/common/techmap.v)).
This path was manually added because the yosys binary couldn't find it, but since then our yosys install was fixed and we no longer need this.